moyopy/src/lib.rs

```diff
@@ -6,11 +6,54 @@ pub mod data;
 
 use moyo::base::AngleTolerance;
 use moyo::data::Setting;
-use moyo::MoyoDataset;
+use moyo::{CrystalSystem, MoyoDataset};
 
 use crate::base::{PyMoyoError, PyOperations, PyStructure};
 use crate::data::{operations_from_number, PyCentering, PyHallSymbolEntry, PySetting};
 
+/// The crystal system of a space group.
+#[derive(Debug, PartialEq, Eq)]
+#[pyclass(name = "CrystalSystem", frozen)]
+pub enum PyCrystalSystem {
+    Triclinic,
+    Monoclinic,
+    Orthorhombic,
+    Tetragonal,
+    Trigonal,
+    Hexagonal,
+    Cubic,
+}
+
+impl From<CrystalSystem> for PyCrystalSystem {
+    fn from(cs: CrystalSystem) -> Self {
+        match cs {
+            CrystalSystem::Triclinic => Self::Triclinic,
+            CrystalSystem::Monoclinic => Self::Monoclinic,
+            CrystalSystem::Orthorhombic => Self::Orthorhombic,
+            CrystalSystem::Tetragonal => Self::Tetragonal,
+            CrystalSystem::Trigonal => Self::Trigonal,
+            CrystalSystem::Hexagonal => Self::Hexagonal,
+            CrystalSystem::Cubic => Self::Cubic,
+        }
+    }
+}
+
+impl std::fmt::Display for PyCrystalSystem {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+#[pymethods]
+impl PyCrystalSystem {
+    fn __str__(&self) -> String {
+        self.to_string()
+    }
+
+    #[classattr]
+    const __module__: &'static str = "moyopy";
+}
```

Comment:
not sure about the code in this file. looks needlessly verbose but couldn't find another way that silences all compiler warnings... maybe due to me not knowing my way round rust yet